wallet/src/de/schildbach/wallet/security/BiometricHelper.kt

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2022 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.security
+
+import android.content.Context
+import android.security.keystore.KeyPermanentlyInvalidatedException
+import androidx.biometric.BiometricManager
+import androidx.biometric.BiometricManager.Authenticators.BIOMETRIC_STRONG
+import androidx.biometric.BiometricPrompt
+import androidx.biometric.BiometricPrompt.CryptoObject
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.FragmentActivity
+import de.schildbach.wallet_test.R
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.suspendCancellableCoroutine
+import org.dash.wallet.common.Configuration
+import org.dash.wallet.common.ui.dialogs.AdaptiveDialog
+import org.slf4j.LoggerFactory
+import java.io.IOException
+import javax.crypto.Cipher
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+open class BiometricException(val errorCode: Int, message: String): Exception(message)
+class BiometricLockoutException(val isPermanent: Boolean, errorCode: Int, message: String) : BiometricException(errorCode, message)
+
+class BiometricHelper(context: Context, private val configuration: Configuration) {
+    companion object {
+        private const val AUTH_TYPES = BIOMETRIC_STRONG
```

Comment:
`BIOMETRIC_STRONG` seem to limit available biometric options to fingerprint scanners. Face recognition doesn't work with this type.